dnsprovider/google_test.go

```diff
@@ -0,0 +1,338 @@
+}
+
+type mockErrManagedZonesListCall struct{}
+
+func (m *mockErrManagedZonesListCall) Pages(ctx context.Context, f func(*dns.ManagedZonesListResponse) error) error {
+	return fmt.Errorf("failed")
+}
+
+type mockManagedZonesClient struct{}
+
+func (m *mockManagedZonesClient) Create(project string, managedzone *dns.ManagedZone) managedZonesCreateCallInterface {
+	return &mockManagedZonesCreateCall{}
+}
+
+func (m *mockManagedZonesClient) Delete(project string, managedZone string) managedZonesDeleteCallInterface {
+	return &mockManagedZonesDeleteCall{}
+}
+
+func (m *mockManagedZonesClient) List(project string) managedZonesListCallInterface {
+	return &mockManagedZonesListCall{}
+}
+
+type mockErrManagedZonesClient struct{}
+
+func (m *mockErrManagedZonesClient) Create(project string, managedzone *dns.ManagedZone) managedZonesCreateCallInterface {
+	return &mockErrManagedZonesCreateCall{}
+}
+
+func (m *mockErrManagedZonesClient) Delete(project string, managedZone string) managedZonesDeleteCallInterface {
+	return &mockErrManagedZonesDeleteCall{}
+}
+
+func (m *mockErrManagedZonesClient) List(project string) managedZonesListCallInterface {
+	return &mockErrManagedZonesListCall{}
+}
+
+type mockResourceRecordSetsListCall struct{}
+
+func (m *mockResourceRecordSetsListCall) Pages(ctx context.Context, f func(*dns.ResourceRecordSetsListResponse) error) error {
+	return f(&dns.ResourceRecordSetsListResponse{Rrsets: expectedRecordSets})
+}
+
+type mockErrResourceRecordSetsListCall struct{}
+
+func (m *mockErrResourceRecordSetsListCall) Pages(ctx context.Context, f func(*dns.ResourceRecordSetsListResponse) error) error {
+	return fmt.Errorf("failed")
+}
+
+type mockResourceRecordSetsClient struct{}
+
+func (m *mockResourceRecordSetsClient) List(project string, managedZone string) resourceRecordSetsListCallInterface {
+	return &mockResourceRecordSetsListCall{}
+}
+
+type mockErrResourceRecordSetsClient struct{}
+
+func (m *mockErrResourceRecordSetsClient) List(project string, managedZone string) resourceRecordSetsListCallInterface {
+	return &mockErrResourceRecordSetsListCall{}
+}
+
+type mockChangesCreateCall struct{}
+
+func (m *mockChangesCreateCall) Do(opts ...googleapi.CallOption) (*dns.Change, error) {
+	return nil, nil
+}
+
+type mockErrChangesCreateCall struct{}
+
+func (m *mockErrChangesCreateCall) Do(opts ...googleapi.CallOption) (*dns.Change, error) {
+	return nil, fmt.Errorf("failed")
+}
+
+type mockChangesClient struct{}
+
+func (m *mockChangesClient) Create(project string, managedZone string, change *dns.Change) changesCreateCallInterface {
+	return &mockChangesCreateCall{}
+}
+
+type mockErrChangesClient struct{}
+
+func (m *mockErrChangesClient) Create(project string, managedZone string, change *dns.Change) changesCreateCallInterface {
+	return &mockErrChangesCreateCall{}
+}
+
+func TestGoogleZones(t *testing.T) {
+	provider := &googleProvider{
+		project:            "project",
+		managedZonesClient: &mockManagedZonesClient{},
```

Comment:
Thanks for the tests. The mocking hides that we're calling `List` with the right project id, for instance. But I think we can add this later if needed.